client/src/components/CommentApp/components/CommentHeader/index.tsx

```diff
@@ -53,6 +54,24 @@ export const CommentHeader: FunctionComponent<CommentHeaderProps> = ({
     }
   };
 
+  const [menuOpen, setMenuOpen] = useState(false);
+  useEffect(() => {
+    if (menuOpen && !focused) {
+      setMenuOpen(false);
+    }
+  }, [focused]);
+  const menuRef = useRef<HTMLDivElement>(null);
+
+  const toggleMenu = (e: React.MouseEvent) => {
+    e.preventDefault();
+    if (!menuOpen) {
+      setMenuOpen(true);
+      setTimeout(() => menuRef.current?.focus(), 1);
```

Comment:
Could use use a `useEffect` for this?

Reply:
For the focus part specifically? Yep, I think that could be moved into the `useEffect` hook. Or the toggling as well?

Reply:
Just the focusing part